app/models/miq_expression/component/leaf.rb

```diff
@@ -0,0 +1,13 @@
+class MiqExpression::Component::Leaf < MiqExpression::Component::Base
+  def self.build(options, timezone)
+    new(MiqExpression::Field.parse(options["field"]), options["value"], timezone)
```

Comment:
wonder if it is possible to have `mode[]` and implement the `less_than` relative date logic in a more generic way.
Not sure how that would restructure this code, but seems like it could reduce the code in a number of components.

This is probably (definitely?) for a future refactor.

After doing that, maybe this "mode" could be passed in and the number of nodes would reduce significantly